google/cloud/sql/connector/connector.py

```diff
@@ -112,5 +115,11 @@ def connect(
         timeout = kwargs["connect_timeout"]
     else:
         timeout = 30  # 30s
-
-    return icm.connect(driver, ip_types, timeout, **kwargs)
+    try:
+        return icm.connect(driver, ip_types, timeout, **kwargs)
+    except TimeoutError as e:
+        raise (e)  # if the error is because of a timeout, let it bubble up
```

Comment:
Are we sure we want an exception for TimeoutError here? 

What if timeout is because the instance's IP has changed, and we need to refresh to get the updated IP?

Reply:
That's a good point. Updated this to force a refresh on any error.